packages/data-broker/src/services/weather/types.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2022 Beyond Essential Systems Pty Ltd
+ */
+
+import { DataElement as BaseDataElement } from '../../types';
+
+export type WeatherProperty = 'min_temp' | 'max_temp' | 'precip';
+
+type WeatherDataSourceConfig = Partial<{
+  weatherForecastData: boolean;
+}>;
+
+export type DataElement = BaseDataElement & {
```

Comment:
Same comment as before about unioned config type.

Reply:
The approach used here has some advantages:

1. Types do not have to be shared/redefined between higher and lower level modules. For example, `DataElement` in `dhis` supports a `dhisDataType` option, which is defined as an exported `type DataType = 'DataElement' | 'Indicator' | 'ProgramIndicator'`.
2. It is easier to see and a glance which options are supported by which service. This is also possible with a union type at the higher level types, it would just be more verbose.

On the flip side (also mentioned in your comment), listing all possible options is a more accurate representation of the actual data source records. There is a tradeoff here, I personally like the current pattern as it feels like type details are defined at the right level.

I'm definitely happy changing this, just wanted to communicate the aspects that influenced the current implementation before doing so. Please let me know what you think :-)

Reply:
Either way seems good to me. Pros and cons as you say. Happy with it as you have it now.